configure.ac

```diff
@@ -219,24 +219,38 @@ AC_CHECK_LIB([gsl],[gsl_sf_bessel_Jn], [],
 # Check for HDF5 library
 AC_ARG_WITH(hdf5, [AC_HELP_STRING([--without-hdf5],[do not allow HDF5 output])], ok=$withval, ok=yes)
 if test "$ok" = "yes"; then
-	AC_CHECK_LIB(z,deflate, [],[AC_MSG_WARN([zlib is required for HDF5!])])
-
-	save_CC=$CC
-	if test "x$with_mpi" = "xyes"; then
-	   CC=$MPICXX
-	fi
-	save_LIBS_0="$LIBS" # save, to check later if we found any library
-	AC_CHECK_LIB(hdf5, H5Pcreate, [
-		AC_CHECK_HEADERS(hdf5.h, [LIBS="-lhdf5 $LIBS"
-                     AC_DEFINE(HAVE_HDF5,1,[Define if we have & link HDF5])])])
-	if test x"$save_LIBS_0" = x"$LIBS"; then
-		AC_MSG_WARN([Couldn't find the HDF5 library!!  Switching to --without-hdf5.])
-	fi
-	CC=$save_CC
-
-        if test "x$with_mpi" = "xyes"; then
-		AC_CHECK_FUNCS(H5Pset_mpi H5Pset_fapl_mpio)
-	fi
+    AC_CHECK_LIB(z,deflate, [],[AC_MSG_WARN([zlib is required for HDF5!])])
+
+    save_CC="$CC"
+    save_CPPFLAGS="$CPPFLAGS"
+    save_LIBS="$LIBS"
+
+    # Attempt to locate HDF5 via pkg-config and fall back to a default.
+    PKG_CHECK_MODULES(HDF5, hdf5, [
+        CPPFLAGS="$CPPFLAGS $HDF5_CFLAGS"
+        LIBS="$HDF5_LIBS $LIBS"
+    ], [
+        LIBS="-lhdf5 $LIBS"
```

Comment:
This isn't right — we can't add `-lhdf5` by default because Meep still compiles even if HDF5 is not present at all.  You have to fall back to the `AC_CHECK_LIB(` test below.

Furthermore, the HDF5 test needs to come *after* you set the MPI compiler, because the parallel version of HDF5 must be linked with MPI in order for linking to succeed.

Reply:
This section performs set up work for the actual check of HDF5 functionality below (via AC_CHECK_LIB and AC_CHECK_HEADER) by setting appropriate LIBS and CPPFLAGS values.

If the actual HDF5 check fails, then LIBS, CPPFLAGS and CC are all reset to their saved values.
If the actual HDF5 check succeeds, then only CC is reset to its saved non-MPI value.